src/components/TradeWidget.jsx

```diff
@@ -1,34 +1,27 @@
 import React, { Component } from 'react';
 import SetTrade from './SetTrade';
 import DoTrade from './DoTrade';
+import WidgetContext from '../contexts/WidgetContext';
+
 
 class TradeWidget extends Component {
   render() {
     return (
-      <div>
+      <WidgetContext.Consumer>
         {
-          this.props.trade.step === 1
-            ?
-            <SetTrade network={this.props.network}
-                      account={this.props.account}
-                      proxy={this.props.proxy}
-                      setMainState={this.props.setMainState}
-                      fasterGasPrice={this.props.fasterGasPrice}
-                      doTrade={this.props.doTrade}
-                      trade={this.props.trade}
-                      balances={this.props.balances} />
-            :
-            <DoTrade network={this.props.network}
-                      account={this.props.account}
-                      proxy={this.props.proxy}
-                      trade={this.props.trade}
-                      transactions={this.props.transactions}
-                      setMainState={this.props.setMainState}
-                      fasterGasPrice={this.props.fasterGasPrice}
-                      reset={this.props.reset}
-                      showTxMessage={this.props.showTxMessage} />
+          values => (
+            <React.Fragment>
+              {
+                values.trade.step === 1
+                  ?
+                  <SetTrade {...values} />
```

Comment:
What if we remove those props from `SetTrade` and `DoTrade` and consume it inside?